osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/config/MetricsConfig.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2022 Alliander N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package org.opensmartgridplatform.shared.config;
+
+import io.micrometer.core.instrument.binder.db.PostgreSQLDatabaseMetrics;
+import io.micrometer.core.instrument.binder.jvm.ClassLoaderMetrics;
+import io.micrometer.core.instrument.binder.jvm.DiskSpaceMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmGcMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmMemoryMetrics;
+import io.micrometer.core.instrument.binder.jvm.JvmThreadMetrics;
+import io.micrometer.core.instrument.binder.logging.LogbackMetrics;
+import io.micrometer.core.instrument.binder.system.ProcessorMetrics;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import java.io.File;
+import java.sql.Connection;
+import java.sql.SQLException;
+import javax.sql.DataSource;
+import org.opensmartgridplatform.shared.application.config.AbstractConfig;
+import org.opensmartgridplatform.shared.metrics.PrometheusMetricsServer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Provides a Prometheus metrics server with optional default metrics for CPU, memory, database. See
+ * {@link PrometheusMetricsServer} for details on how to configure the port/path of the metrics
+ * server.
+ *
+ * <p>The provided {@link PrometheusMeterRegistry} bean can be used to add custom metrics.
+ */
+@Configuration
+// Starts the PrometheusMetricsServer
```

Comment:
I guess this comment had better been removed or altered. I don't understand what it clarifies here, and to which code it applies.
If it is about the component scan making sure the `PrometheusMetricsServer` component is included in the Spring context whenever the `MetricsConfig` is imported, that could be a useful comment, but starting the metrics server is dependent on code in the `PrometheusMetricsServer` and the `PrometheusEnabledCondition`.

Reply:
Clarified this in person; it's for the @ComponentScan below. Changed the comment to be more clear